pilz_trajectory_generation/include/pilz_trajectory_generation/robot_trajectory_helper.h

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (c) 2018 Pilz GmbH & Co. KG
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef ROBOT_TRAJECTORY_HELPER_H
+#define ROBOT_TRAJECTORY_HELPER_H
+
+#include <moveit/robot_trajectory/robot_trajectory.h>
+
+namespace pilz_trajectory_generation
+{
+
+class RobotTrajectoryHelper
```

Comment:
Instead of introducing another helper class, I suggest working towards an upcoming refactoring. Like I see it, we have three different types of software components in the pilz_trajectory_generation package:
* Firstly, we have the "pure" trajectory generators which take a `MotionPlanRequest` and generate a "simple" linear, cirucular or ptp trajectory.
* Secondly, we have software components which I suggest calling `TrajectoryMerger`. The `TrajectoryBlender` class and your newly created `RobotTrajectoryHelper` belong to this type of software component. Simply put: They take two trajectories and merge them together (one way or another).
* Thirdly, we have the software components which basically use the trajectory generators and trajectory mergers to actually do something. An example for these software components is the `CommandListManager`.
Based on these considerations, I suggest to introduce a base class called `TrajectoryMerger` and derive a specific TrajectoryMerger which basically contains the functionality which you put into your `RobotTrajectoryHelper` class.

Reply:
I can see your point and followed your suggestion. Can you please review the new commit?